cloudify_awssdk/s3/resources/bucket_object.py

```diff
@@ -0,0 +1,256 @@
+        self.client.delete_object(**params)
+
+
+def _download_remote_file(file_url):
+    """
+    Try to download the file provided by the blueprint so because
+    execution directory on a manager is not the blueprint root and want to
+    make sure file is accessible by manager context
+    :param file_url: file path under blueprint root,
+    e.g: examples/s3-feature-demo/local-s3-object.txt
+    :return: file path which can be accessed by cloudify manager
+    """
+    # Make temp file to write file to
+    _, file_path = tempfile.mkstemp()
+
+    # Try to download and read file url
+    target_file = urllib.urlopen(file_url)
+    # Open "file_path" for write
+    fh = open(file_path, 'w')
+
+    # Read data from target file and write to the "file_path"
+    fh.write(target_file.read())
+
+    # Close file cursor
+    fh.close()
+
+    # Return file_path to be uploaded
+    return file_path
+
+
+def _download_local_file(local_path):
+    try:
+        path = ctx.download_resource(local_path)
+    except HttpException:
+        raise NonRecoverableError(
+            '{} file does not exist.'.format(local_path))
+    return path
+
+
+@decorators.aws_resource(resource_type=RESOURCE_TYPE)
+def prepare(ctx, resource_config, **_):
+    """Prepares an AWS S3 Bucket Object"""
+    # Save the parameters
+    ctx.instance.runtime_properties['resource_config'] = resource_config
+
+
+@decorators.check_swift_resource
+@decorators.aws_resource(S3BucketObject, RESOURCE_TYPE)
+def create(ctx, iface, resource_config, **_):
+    """Creates an AWS S3 Bucket Object"""
+
+    # Create a copy of the resource config for clean manipulation.
+    params = \
+        dict() if not resource_config else resource_config.copy()
+
+    # Get the bucket object key from params
+    object_key = params.get(OBJECT_KEY)
+    if not object_key:
+        raise NonRecoverableError('{} param is required'.format(OBJECT_KEY))
+
+    utils.update_resource_id(ctx.instance, object_key)
+    source_type = ctx.node.properties.get(OBJECT_SOURCE_TYPE)
+
+    cloudify_path = None
+    object_body = None
+
+    # If "source_type" is either local or remote then we need to download
+    # the file from remote or local directory and then parse the data as
+    # bytes and prepared it to be sent on the "Body" param for "put_object"
+    # method
+    if source_type in [OBJECT_LOCAL_SOURCE, OBJECT_REMOTE_SOURCE]:
+        path = ctx.node.properties.get(OBJECT_PATH)
+        if not path:
+            raise NonRecoverableError(
+                'path param must be provided when '
+                'source_type is selected as remote or local')
+
+        if source_type == OBJECT_LOCAL_SOURCE:
+            cloudify_path = _download_local_file(path)
+
+        elif source_type == OBJECT_REMOTE_SOURCE:
+            cloudify_path = _download_remote_file(path)
+        try:
+            object_body = open(cloudify_path, 'rb')
+        except IOError as error:
+            raise NonRecoverableError(
```

Comment:
I see that you print the `strerror`, but I think it'd be safer and would contain more information if you uses the `causes` feature.

Reply:
Done